security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -106,12 +106,12 @@ func (p Plugin) ExchangeToken(ctx context.Context, credFetcher security.CredFetc
 	respData := &federatedTokenResponse{}
 	if err := json.Unmarshal(body, respData); err != nil {
 		return "", time.Now(), resp.StatusCode, fmt.Errorf(
-			"failed to unmarshal response data. HTTP status: %s. Error: %v. Body size: %d", resp.Status, err, len(body))
+			"failed to unmarshal response data. HTTP status: %s. Req: %s. Error: %v. Body size: %d", resp.Status, string(jsonStr), err, len(body))
 	}
 
 	if respData.AccessToken == "" {
 		return "", time.Now(), resp.StatusCode, fmt.Errorf(
-			"exchanged empty token. HTTP status: %s. Response: %v", resp.Status, string(body))
+			"exchanged empty token. HTTP status: %s. Req: %s Response: %v", resp.Status, string(jsonStr), string(body))
```

Comment:
Same here. Don't print out the entire request string.

Reply:
Debugging anything related to GCP is close to impossible at the moment - having the request string helps understand why the request failed.